### ISSUE TITLE
Dispatch rules not working well for <any>

### ISSUE THREAD

**Issue by Kedaqusa1604** [livekit/sip#213]
Hi there. This is a continuation of this issue: [197](https://github.com/livekit/sip/issues/197)

I'm facing some issues because of the way that I used to solve this, creating a single dispatch rule with no trunks  (so, in theory, it should handle any "scenario" or "number")

few weeks ago it was working as expected, but today that I tried again this scenario, all the calls are being rejected by the server.

am I missing any configuration or something? 
this is what my inbound trunks look like:

![{FC762B6B-56CE-48F1-B91A-17BBAA397EBC}](https://github.com/user-attachments/assets/161bec91-2ed0-4a36-a744-428c7c5cadb2)

and these are my dispatch rules:

![{C0888417-44AB-44C6-BA6D-9E8C88839B8C}](https://github.com/user-attachments/assets/1fc93d9b-6ccb-483e-9073-b98d3594b371)

Room for call rule:

```json
{
    "name": "Room for call number",
    "trunk_ids": [],
    "hide_phone_number": false,
    "rule": {
        "dispatchRuleCallee": {
            "roomPrefix": "number-",
            "randomize": false,
            "pin": ""
        }
    }
}
```
thanks in advance for any help.

_Originally posted by @Kedaqusa1604 in https://github.com/livekit/sip/issues/197#issuecomment-2430461188_

**Comment by dennwc** [comment]
Yes, we now require an inbound trunk with no numbers to handle this case. However, the latest LiveKit version doesn't allow creating these trunks. This PR will fix it: https://github.com/livekit/protocol/pull/890.
